types/budget.d.ts

```diff
@@ -40,7 +40,7 @@ declare global {
       export type TimingMetric = 'first-contentful-paint' | 'first-cpu-idle' | 'interactive' | 'first-meaningful-paint' | 'estimated-input-latency';
 
       /** Supported resource types. */
-      export type ResourceType = 'stylesheet' | 'image' | 'media' | 'font' | 'script' | 'document' | 'other';
+      export type ResourceType = 'stylesheet' | 'image' | 'media' | 'font' | 'script' | 'document' | 'other' | 'total' | 'third-party';
```

Comment:
oops, a bit weird that my feedback forced this sorry 😬

I guess if we're being totally accurate it'd be something like 

```ts
export type ResourceType = ...
export type BudgetType = ResourceType | 'total' | 'third-party'
```

?

I'll defer for final decision since it seems like anywhere one might use a `ResourceType`, 'total' might be an option 🤷‍♂

Reply:
I like this idea. How about `ResourceType` and `BudgetResourceType`? `BudgetResourceType` is a bit verbose, but I'm not sure if `BudgetType` would become confusing when timing budgets are likely introduced in the future.

Reply:
`BudgetResourceType` sounds great! 👍 

I knew you'd have some extra future budget knowledge to inject there 😃